### PR TITLE
feat: implement rediscovery

### LINKED ISSUE #71: Implement re-discovery based on serial number
# 🚀 Feature Request

### Description [**REQUIRED**]
When the discovery code detects a device has changed paths on the system, update the device info to match. Can possibly be integrated with `driver.RefreshExistingDevicePaths()`

Currently it looks like the code just skips the device, but does not update the info.

### PR DESCRIPTION
**Pull request by presatish**
**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [X] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: docs/advanced-options.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 [Video Options](#video-options)  
 [Dynamic Discovery](#dynamic-discovery)  
 [Camera Paths](#keep-the-paths-of-existing-camera-up-to-date)
```
Comment (brianointel) [inline docs/advanced-options.md]: 2 spaces need to be added after this line for `RTSP Server` to move down a line.

**Review by brianointel — DISMISSED** [review]
LGTM

**Review by ChristianDarr-personal — DISMISSED** [review]
LGTM

**Review by ChristianDarr-personal — APPROVED** [review]

**Review by lenny-goodell — APPROVED** [review]
LGTM

**Pull request merged**